test/functional/api/v1/rubygems_controller_test.rb

```diff
@@ -309,19 +309,19 @@ def self.should_respond_to(format)
       end
     end
 
-    # context "On POST to create with a protected gem name" do
-    #   setup do
-    #     above_downloads_thres = GemTypo::DOWNLOADS_THRESHOLD + 1
-    #     create(:rubygem, name: "best", downloads: above_downloads_thres)
-    #     post :create, body: gem_file("test-1.0.0.gem").read
-    #   end
-
-    #   should respond_with :forbidden
-    #   should "not register new gem" do
-    #     assert_equal 1, Rubygem.count
-    #     assert_equal "There was a problem saving your gem: Name 'test' is too close to typo-protected gem: best", @response.body
-    #   end
-    # end
+    context "On POST to create with an underscore or dash variant of an existing gem" do
+      setup do
+        existing = create(:rubygem, name: "t_es-t", downloads: 3002)
+        existing.versions.create(number: "1.0.0", platform: "ruby")
+        post :create, body: gem_file("test-1.0.0.gem").read
+      end
+
+      should respond_with :forbidden
+      should "not register new gem" do
+        assert_equal 1, Rubygem.count
+        assert_equal "There was a problem saving your gem: Name Your gem 'test' is too similar to an existing gem named 't_es-t'", @response.body
```

Comment:
Can we please drop `Your gem`. Following reads better:
> Name 'test' is too similar to an existing gem named 't_es-t'

Reply:
Fixed on branch, thank you!